pkg/nfd-master/nfd-master.go

```diff
@@ -605,20 +604,85 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// filterExtendedResources filters extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) filterExtendedResources(features *nfdv1alpha1.Features, extendedResources ExtendedResources) ExtendedResources {
+	outExtendedResources := ExtendedResources{}
+	deniedNs := map[string]struct{}{"kubernetes.io": {}}
+	deniedWildCarNs := map[string]struct{}{".kubernetes.io": {}}
+	for extendedResource, capacity := range extendedResources {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, deniedWildCarNs, deniedNs) {
+			if ns != nfdv1alpha1.ExtendedResourceNs && ns != nfdv1alpha1.FeatureLabelNs &&
+				!strings.HasSuffix(ns, nfdv1alpha1.ExtendedResourceSubNsSuffix) && !strings.HasSuffix(ns, nfdv1alpha1.FeatureLabelSubNsSuffix) {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
+				continue
+			}
+		}
+
+		// Dynamic Value
+		if strings.HasPrefix(capacity, "@") {
+			// Remove the @ prefix
+			capacity = strings.TrimPrefix(capacity, "@")
+			// capacity is a string in the form of attribute.featureset.elements
+			split := strings.SplitN(capacity, ".", 3)
+			featureName := split[0] + "." + split[1]
+			elementName := split[2]
+			attrFeatureSet, ok := features.Attributes[featureName]
+			if !ok {
+				klog.Errorf("feature %s not found. Ignoring Extended Resource %q/n", featureName, extendedResource)
```

Comment:
Invalid linefeed, and we don't need linefeeds anyway

```suggestion
				klog.Errorf("feature %s not found. Ignoring Extended Resource %q", featureName, extendedResource)
```